source/Calamari.Shared/Integration/Certificates/NoOpWindowsX509CertificateStore.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Calamari.Integration.Certificates
+{
+    /// <summary>
+    /// Stand in replacement for IWindowsX509CertificateStore that will be registered for non Windows machines.
+    /// This should never end up being called. If it is, something has gone wrong somewhere else
+    /// </summary>
+    public class NoOpWindowsX509CertificateStore: IWindowsX509CertificateStore
```

Comment:
In future we wont want to rely on compiler directives to prevent parts of the code from not compiling. This class will be registered for non-windows calamari.